src/com/lightdev/app/shtm/CombinedAttribute.java

```diff
@@ -148,18 +148,11 @@ else if (key.equals(CSS.Attribute.MARGIN)) {
      * @param a  the set of attributes to get the value from
      */
     private void setValue(final int side, final Object key, final AttributeSet a) {
-        if ((includeParents) || ((!includeParents) && (aNames.contains(key)))) { // filter out parent attributes
+        if (includeParents || aNames.contains(key)) { // filter out parent attributes
```

Comment:
Why have you changed the logic in the expression?

Reply:
When **(includeParents)** is **false**, the second expression of the or is executed. In this case **(!includeParents)** will always be **true** so it can be removed

Reply:
present[side] = true; is in both if/else branches, so it can be extracted. Once we do that, the following expression can replace the if/else
attr != null ? attr.toString() : defaultValue(attributeKey);